### ISSUE TITLE
Activity not retried as expected during Activity failure

### ISSUE THREAD

**Issue by olitomlinson** [Azure/azure-functions-durable-extension#769]
We use `context.CallactivityWithRetryAsync` API to call an activity multiple times in the event of a failure/exception with an Activity.

In this particular instance, the Activity was only called once, (the initial call) and it failed, and then no retry attempts were made by the Orchestration. The Orchestration then entered the Failed state.

I believe this happened because I see no rows in the History table that would typically show me the retries being attempted when an activity failure has occurred.

This is the Retry configuration in use

```
            var retryOptions = new RetryOptions(TimeSpan.FromSeconds(5), 10);
            retryOptions.BackoffCoefficient = 2;
            retryOptions.MaxRetryInterval = TimeSpan.FromSeconds(60);
            retryOptions.RetryTimeout = TimeSpan.FromMinutes(10);
```

**Function App Info**

UK South
DF 1.8.0 (azure Functions V2)  - Consumption
Excecution Id : ffb3e67afdb241cb8d3f0b00beb28626
Orchestration Id : 9802e54c-8191-4ece-9c9b-cf854e251617

**Other info**

The function App was under load at the time when this occurred (starting and completing approx 40,000 short-lived orchestrations)

The function App was also unexpectedly and unusually experiencing thousands of Service Bus receive errors (which plays an integral part in how we receive `external events` into our Orchestrations).

It may be possible that there was something bad occurring in the underlying host which may have contributed to this issue somehow.

**Comment by cgillum** [comment]
Is the orchestration ID mentioned above correct? I don't see any orchestration instances matching this ID in UK South over the last 7 days.

**Comment by olitomlinson** [comment]
Sorry, this was from the 9th of May, around 15:30 UTC.

**Comment by cgillum** [comment]
Thanks. Here is what our telemetry is showing me (you should have access to the same info in Application Insights, if that's enabled for you):

1. At **2019-05-09 15:30:29**, the function **Router_GetAvailableEdges_ActivityTrigger** failed with an IOException on a socket operation.
1. At **2019-05-09 15:40:49** (over 10 minutes later), the orchestration replayed. Since you configured a,10 minute `RetryTimeout` value, I suspect the retry logic decided not to retry this operation, so the orchestration failed.

Looking more closely at our storage telemetry, it appears your `taskhub-control-15` queue was severely backed up (possibly related to the Service Bus issues you mentioned, or some other perf issue). In fact, there was a 10+ minute gap between the time the activity function failed and when the failure message got picked up by the orchestration. This explains the previous finding.

So to summarize, I think a perf issue in the app caused a significant delay in orchestration processing, and this delay caused you to miss your configured retry window. Possible workarounds would include extending the retry timeout to something larger, or perhaps not using one at all.

**Comment by olitomlinson** [comment]
Thanks @cgillum that makes sense!
